client.go

```diff
@@ -0,0 +1,96 @@
+package skynet
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"gitlab.com/NebulousLabs/errors"
+)
+
+type (
+	// SkynetClient is the Skynet Client which can be used to access Skynet.
+	SkynetClient struct {
+		PortalURL string
+		Options   Options
+	}
+
+	// requestOptions contains the options for a request.
+	requestOptions struct {
+		Options
+
+		method    string
+		reqBody   io.Reader
+		extraPath string
+		query     url.Values
+	}
+)
+
+// New creates a new Skynet Client which can be used to access Skynet.
+func New() SkynetClient {
+	return NewCustom("", Options{})
+}
+
+// NewCustom creates a new Skynet Client with a custom portal URL and options.
+// Pass in "" for the portal to let the function select one for you.
+func NewCustom(portalURL string, customOptions Options) SkynetClient {
+	if portalURL == "" {
+		portalURL = DefaultPortalURL
```

Comment:
since this is exported, I think it is fine to require devs to pass in the portal vs accepting an empty string. Then we can add some input validation here around supplying empty inputs. I understand allowing the blank string the first time around but now that there is the `New()` I think we should enforce a supplied portal here.

Reply:
Was there a reason other that reducing friction that you had originally through of allowing the blank string?